server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -44,9 +44,21 @@ interface ImmichTags extends Tags {
   MediaGroupUUID?: string;
 }
 
-const exifDate = (dt: ExifDateTime | string | undefined) => (dt instanceof ExifDateTime ? dt?.toDate() : null);
-const validate = <T>(value: T): T | null => (typeof value === 'string' ? null : value ?? null);
+type ExifEntityWithoutGeocodeAndTypeOrm = Omit<
+  ExifEntity,
+  'city' | 'state' | 'country' | 'description' | 'exifTextSearchableColumn'
+>;
 
+const exifDate = (dt: ExifDateTime | string | undefined) => (dt instanceof ExifDateTime ? dt?.toDate() : null);
+const validate = <T>(value: T): NonNullable<T> | null => {
+  if (typeof value === 'string' && value === '') {
+    return null;
+  }
+  if (typeof value === 'number' && isNaN(value)) {
+    return null;
+  }
+  return value ?? null;
```

Comment:
If the value is a string, it means it failed to parse it as a number, so any string should return null.
If the value is a number, we should probably check `isNaN` and `isFinite`.

Reply:
I was curious about that. I think I changed this early on in the solution, since `tags.FocalLength` was still reporting as `string`.